openshift/template.yaml

```diff
@@ -373,3 +428,70 @@ objects:
     selector:
       matchLabels:
         app: assisted-service
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: assisted-service-envoy-config
+data:
+  main.yaml: |
+    admin:
+      access_log_path: /dev/null
+      address:
+        socket_address: { address: 0.0.0.0, port_value: 9000 }
+
+    static_resources:
+      listeners:
+      - name: ingress
+        address:
+          socket_address: { address: 0.0.0.0, port_value: ${{ENVOY_PORT}} }
+        filter_chains:
+        - filters:
+          - name: envoy.filters.network.http_connection_manager
+            typed_config:
+              "@type": type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
+              access_log:
+              - name: envoy.access_loggers.file
+                typed_config:
+                  "@type": type.googleapis.com/envoy.extensions.access_loggers.file.v3.FileAccessLog
+                  path: /dev/stdout
+                  jsonFormat:
+                    authority: "%REQ(:AUTHORITY)%"
+                    duration: "%DURATION%"
+                    path: "%REQ(X-ENVOY-ORIGINAL-PATH?:PATH)%"
+                    protocol: "%PROTOCOL%"
+                    response_code: "%RESPONSE_CODE%"
+                    response_flags: "%RESPONSE_FLAGS%"
+                    route_name: "%ROUTE_NAME%"
+                    upstream_host: "%UPSTREAM_HOST%"
+                    user_agent: "%REQ(USER-AGENT)%"
+                    x_agent_name: "%REQ(X-AGENTNAME)%"
+                    x_forwarded_for: "%REQ(X-FORWARDED-FOR)%"
+              stat_prefix: ingress_http
+              codec_type: AUTO
+              route_config:
+                name: local_route
+                virtual_hosts:
+                - name: local_service
+                  domains: ["*"]
+                  routes:
+                  - match: { prefix: "/" }
+                    route: { cluster: assisted_service }
+              http_filters:
+              - name: envoy.filters.http.router
+                typed_config:
+                  "@type": type.googleapis.com/envoy.extensions.filters.http.router.v3.Router
+      clusters:
+      - name: assisted_service
+        connect_timeout: 1s
+        type: STATIC
+        lb_policy: ROUND_ROBIN
```

Comment:
You will probably want to use HTTP/2 here:

```
http2_protocol_options: {}
```

It needs to be enabled explicitly, otherwise all the communication with the backend will use HTTP/1.1, which is less efficient and requires more connections.

Reply:
It looks like the assisted-service is not handling http2 without TLS (golang http library).
However I will try to configure it so that envoy can handle http2 (and terminate it in this case), so it will be easier to switch full http2 from the app.

Reply:
Yes, that is what I mentioned in the overall review comment: currently we don't support Unix sockets or HTTP/2 without TLS; would be nice to have in the future.

Reply:
In the listener side you don't need to configure HTTP/2 explicitly: the default connection manager supports both HTTP/1.1 and HTTP/2 out of the box. But the client (the OpenShift router in this case) will not even try connect using HTTP/2 because it doesn't support _h2c_ (HTTP/2 without TLS) and our service doesn't use TLS.

Reply:
Right, so I take it that once we change the service to handle http2, it will work with http2 out of the box (once added the above mentioned settings), no need to force upgrade right?
If I add such settings right now it would break with "protocol error".

Reply:
Yes, correct.